daemons/ipa-slapi-plugins/ipa-extdom-extop/back_extdom_sss_idmap.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2013-2017 Red Hat, Inc.
+ *
+ * This Program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This Program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this Program; if not, write to the
+ *
+ *   Free Software Foundation, Inc.
+ *   59 Temple Place, Suite 330
+ *   Boston, MA 02111-1307 USA
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <sys/types.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <errno.h>
+#include <pwd.h>
+#include <grp.h>
+#include "back_extdom.h"
+
+/* SSSD only exposes *_timeout() variants if the following symbol is defined */
+#define IPA_389DS_PLUGIN_HELPER_CALLS
+#include <sss_nss_idmap.h>
+
+struct nss_ops_ctx {
+	unsigned int timeout;
+};
+
+static enum nss_status __convert_sss_nss2nss_status(int errcode) {
+	enum nss_status ret = NSS_STATUS_UNAVAIL;
+
+	if (errcode == 0) {
+		ret = NSS_STATUS_SUCCESS;
+	} else if (errcode == ENOENT) {
+		ret = NSS_STATUS_NOTFOUND;
+	} else if(errcode == ERANGE) {
+		ret = NSS_STATUS_TRYAGAIN;
+	} else if(errcode == ETIMEDOUT) {
+		ret = NSS_STATUS_UNAVAIL;
+	} else if(errcode == ETIME) {
+		ret = NSS_STATUS_TRYAGAIN;
+	};
+
+	return ret;
+}
+
+int back_extdom_init_context(struct nss_ops_ctx **nss_context)
+{
+	struct nss_ops_ctx *ctx = NULL;
+
+	if (nss_context == NULL) {
+		return -1;
+	}
+
+	ctx = calloc(1, sizeof(struct nss_ops_ctx));
+
+	if (ctx == NULL) {
+		return ENOMEM;
+	}
+	*nss_context = ctx;
+	return 0;
+}
+
+void back_extdom_free_context(struct nss_ops_ctx **nss_context)
+{
+	if ((nss_context == NULL) || (*nss_context == NULL)) {
+		return;
+	}
+
+	free((*nss_context));
+	*nss_context = NULL;
+}
+
+
+void back_extdom_set_timeout(struct nss_ops_ctx *nss_context,
+			     unsigned int timeout) {
+	if (nss_context == NULL) {
+		return;
+	}
+
+	nss_context->timeout = timeout;
+}
+
+/* TODO: handle buffers and memory allocation in this function */
```

Comment:
what buffers need to be handled ?

Reply:
there is another one as well

Reply:
I removed this todo: after a rewrite of the initial version it became irrelevant.